utils/git.go

```diff
@@ -64,14 +61,75 @@ type CustomTemplates struct {
 	pullRequestTitleTemplate string
 }
 
-func NewGitManager(username, token string, gitParams *Git, dryRun bool, clonedRepoPath string) (*GitManager, error) {
+func NewGitManager() *GitManager {
 	setGoGitCustomClient()
-	basicAuth := toBasicAuth(token, username)
-	templates, err := loadCustomTemplates(gitParams.CommitMessageTemplate, gitParams.BranchNameTemplate, gitParams.PullRequestTitleTemplate)
+	return &GitManager{}
+}
+
+func (gm *GitManager) SetAuth(username, token string) *GitManager {
+	gm.auth = toBasicAuth(token, username)
+	return gm
+}
+
+func (gm *GitManager) SetRemoteGitUrl(remoteHttpsGitUrl string) (*GitManager, error) {
+	// Check if the .git directory exists
+	dotGitExists, err := fileutils.IsDirExists(git.GitDirName, false)
 	if err != nil {
 		return nil, err
 	}
-	return &GitManager{dryRunRepoPath: clonedRepoPath, remoteName: "origin", auth: basicAuth, dryRun: dryRun, customTemplates: templates, git: gitParams}, nil
+
+	if !dotGitExists {
+		// If .git directory doesn't exist, create it with the given remote URL
+		gm.remoteGitUrl = remoteHttpsGitUrl
+		err = vcsutils.CreateDotGitFolderWithRemote(".", vcsutils.RemoteName, remoteHttpsGitUrl)
+		return gm, err
+	}
+
+	if gm.repository == nil {
+		if _, err = gm.SetLocalRepository(); err != nil {
+			return gm, err
+		}
+	}
+
+	gitRemote, err := gm.repository.Remote(gm.remoteName)
+	if err != nil {
+		return nil, fmt.Errorf("'git remote %s' failed with error: %s", gm.remoteName, err.Error())
+	}
+
+	if len(gitRemote.Config().URLs) < 1 {
+		return nil, errors.New("failed to find git remote URL")
+	}
+
+	gm.remoteGitUrl = gitRemote.Config().URLs[0]
```

Comment:
what happens if url > 1 ?

Reply:
The first one should be the URL from which the repository was cloned. Other URLs might represent the forks, etc.

Reply:
@sverdlov93